docs: add guide on using Cilium to secure CloudNativePG traffic

zekiyeaydemir opened the pull request:
This contribution adds a new documentation section on securing CloudNativePG-managed PostgreSQL clusters using Cilium, a CNCF Sandbox project that provides eBPF-powered networking and security. It includes an introduction to Cilium, an example CiliumNetworkPolicy manifest to restrict access to PostgreSQL. 

Closes #7691

zekiyeaydemir committed:
docs: add guide on using Cilium to secure CloudNativePG traffic

Closes #7691

Signed-off-by: Zekiye Aydemir <aydemirzekiye@gmail.com>

zekiyeaydemir committed:
Update securing_cloudnative-pg_network.md

Update Cilium's CNCF graduated status in introduction and fix title

Signed-off-by: Zekiye Aydemir <aydemirzekiye@gmail.com>

zekiyeaydemir committed:
Update securing_cloudnative-pg_network.md

improve wording of Cilium's CNCF status in intro by adding historical context

Signed-off-by: Zekiye Aydemir <aydemirzekiye@gmail.com>

zekiyeaydemir committed:
docs: fix Kernel casing, add networkpolicy examples, and recommend policy editor

- Fixed capitalization of "Kernel" in the introduction
- Added example: allow access from role=backend pods in any namespace
- Added example: allow access from any pod in specific namespace
- Added example: allow access from any pod in the cluster 
- Suggested editor.networkpolicy.io for visual policy creation and validation

Signed-off-by: Zekiye Aydemir <aydemirzekiye@gmail.com>

sxd commented:
I´m working on adding a couple of stuff here @zekiyeaydemir hopefully will finish tomorrow and I can get a review from you will be awesome! Will let you know!

sxd committed:
chore: review

Signed-off-by: Jonathan Gonzalez V. <jonathan.gonzalez@enterprisedb.com>

sxd committed:
chore: review 2

Signed-off-by: Jonathan Gonzalez V. <jonathan.gonzalez@enterprisedb.com>

sxd committed:
lint

Signed-off-by: Jonathan Gonzalez V. <jonathan.gonzalez@enterprisedb.com>

zekiyeaydemir committed:
docs: add matchLabels explanation to cnpg-policy section

Added short explanation about how matchLabels works in the policy example, mentioned And logic.

Signed-off-by: Zekiye Aydemir <aydemirzekiye@gmail.com>

jbattiato committed:
docs: add link to cilium installation and review

Signed-off-by: Jonathan Battiato <jonathan.battiato@enterprisedb.com>

jbattiato committed:
docs: fix link

Signed-off-by: Jonathan Battiato <jonathan.battiato@enterprisedb.com>

mnencia commented:
/ok-to-merge docs only
